src/pickups/datastore/pickups.js

```diff
@@ -31,13 +42,15 @@ export default {
         group,
         collectors: pickup.collectorIds.map(rootGetters['users/get']),
         feedbackGivenBy: pickup.feedbackGivenBy ? pickup.feedbackGivenBy.map(rootGetters['users/get']) : [],
+        hasStarted: pickup.date <= new Date(),
         ...metaStatusesWithId(getters, ['save', 'join', 'leave'], pickup.id),
       }
     },
     upcoming: (state, getters) => {
       return Object.values(state.entries)
-        .filter(p => p.date >= state.now)
+        .filter(p => p.date >= subMinutes(state.now, pickupRunningTime))
         .map(getters.enrich)
+        .filter(p => p.date >= state.now || (p.isUserMember && p.date <= state.now))
```

Comment:
you could use `p.hasStarted` here to make it easier to understand:

```suggestion
        .filter(p => !p.hasStarted || (p.isUserMember && p.hasStarted))
```